packages/zoe/src/contracts/otcDesk.js

```diff
@@ -0,0 +1,117 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+import { trade } from '../contractSupport';
+
+import '../../exported';
+
+/**
+ * The creator gets a creatorFacet that allows them to add inventory, make quotes and
+ * create specific invitations to exercise those quotes.
+ * @type {ContractStartFn}
+ */
+const start = zcf => {
+  const { zcfSeat: marketMakerSeat } = zcf.makeEmptySeatKit();
+
+  const makeQuote = (
+    userGiveQuote,
+    userWantQuote,
+    timerAuthority,
+    deadline,
+  ) => {
+    let quoteActive = true;
+    E(timerAuthority).setWakeup(
+      deadline,
+      harden({ wake: () => (quoteActive = false) }),
+    );
+    const acceptQuote = seat => {
+      assert(quoteActive, details`The quote was no longer active`);
+      const { want: userWantActual, give: userGiveActual } = seat.getProposal();
```

Comment:
This doesn't seem to match the model in the linked article. I thought the main point of the signed price quotes was that they were intended to be commitments to trade at the stated price. This contract will trade at that price if it has remaining funds, but there's no commitment to the counterparty that the funds are available.

Reply:
Really good point. Fixed